bot/interact.go

```diff
@@ -0,0 +1,162 @@
+package bot
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/bcneng/candebot/slackx"
+
+	"github.com/bcneng/candebot/cmd"
+	"github.com/slack-go/slack"
+)
+
+var urlRegex = regexp.MustCompile(`(?mi)^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$`)
```

Comment:
consider parsing the URL to validate it with https://golang.org/pkg/net/url/#URL.Parse